simtools/visualization/visualize.py

```diff
@@ -531,3 +542,137 @@ def plot_hist_2D(data, **kwargs):
     fig.tight_layout()
 
     return fig
+
+
+@u.quantity_input(x=u.m, y=u.m, radius=u.m)
+def get_telescope_patch(name, x, y, radius):
+    """
+    Collect the patch of one telescope to be plotted by plot_array.
+
+    Parameters
+    ----------
+    name: str
+        Name of the telescope (type).
+    x: astropy.units.m
+        x position of the telescope in meters.
+    y: astropy.units.m
+        y position of the telescope in meters.
+    radius: astropy.units.m
+        Radius of the telescope sphere in meters.
+
+    Returns
+    -------
+    patch
+        Instance of mpatches.Circle.
+    """
+    tel_obj = leg_h.TelescopeHandler()
+    valid_name = names.get_telescope_type(name)
+    fill_flag = False
+    if valid_name == mst:
+        fill_flag = True
+    if valid_name == sct:
+        patch = mpatches.Rectangle(
+            ((x - radius / 2).value, (y - radius / 2).value),
+            width=radius.value,
+            height=radius.value,
+            fill=False,
+            color=tel_obj.colors_dict[sct],
+        )
+    else:
+        patch = mpatches.Circle(
+            (x.value, y.value),
+            radius=radius.value,
+            fill=fill_flag,
+            color=tel_obj.colors_dict[valid_name],
+        )
+    return patch
+
+
+@u.quantity_input(rotate_angle=u.deg)
+def plot_array(telescopes, rotate_angle=0 * u.deg):
+    """
+    Plot the array of telescopes.
+
+    Parameters
+    ----------
+    telescopes: dict
+        Dictionary with the telescope position and names.
+    rotate_angle:
+        Angle to rotate the plot. For rotate_angle = 0 the x-axis points towards the east, and\
+        the y-axis points towards the North.
+
+    Returns
+    -------
+    plt.figure
+        Instance of plt.figure with the array of telescopes plotted.
+
+    """
+
+    fig, ax = plt.subplots(1)
+    legend_objects = []
+    legend_labels = []
+    tel_counters = {one_telescope: 0 for one_telescope in names.all_telescope_class_names}
+    if rotate_angle != 0:
+        telescopes["pos_x"], telescopes["pos_y"] = gen.rotate(
```

Comment:
Does this modify the original positions in the telescope variables? Meaning calling `plot_array` twice with a non-zero rotation angles will result in different plots?

Reply:
I tested it now and indeed this happens. do you think that it is a problem? 
what about introducing a flag to ask if we want to permanently rotate or just for the plot?

Reply:
Yes, we will have forget soon that a plotting function is modifying the values. So better to copy it into new variables.